docs/advanced/document-store/opensearch.md

```diff
@@ -0,0 +1,414 @@
+
+  Length                 2000
+  Homogenous Documents   True
+  Common Attributes      ('id', 'embedding')
+
+                      Attributes Summary
+
+  Attribute   Data type      #Unique values   Has empty value
+ ─────────────────────────────────────────────────────────────
+  embedding   ('ndarray',)   1000             False
+  id          ('str',)       1000             False
+
+              Storage Summary
+
+  Backend            OpenSearch
+  Host               http://localhost:9200
+  Distance           cosine
+  Vector dimension   128
+  ES config          {}
+
+[0.14890289 0.3168339  0.03050802 0.06785086 0.94719299 0.32490566
+ ...]
+```
+
+Other functions behave the same as in-memory DocumentArray.
+
+### Bulk request customization
+
+You can customize how bulk requests is being sent to Opensearch when adding documents by adding additional `kwargs` on `extend` method call. See [the official Documentation](https://opensearch.org/docs/1.2/opensearch/rest-api/document-apis/bulk/) for more details. See the following code for example:
+
+```python
+from docarray import Document, DocumentArray
+import numpy as np
+
+n_dim = 3
+
+da = DocumentArray(
+    storage='opensearch',
+    config={'n_dim': 3, 'columns': {'price': 'int'}, 'distance': 'l2'},
+)
+
+with da:
+    da.extend(
+        [
+            Document(id=f'r{i}', embedding=i * np.ones(n_dim), tags={'price': i})
+            for i in range(10)
+        ],
+        thread_count=4,
+        chunk_size=500,
+        max_chunk_bytes=104857600,
+        queue_size=4,
+    )
+```
+
+````{admonition} Note
+:class: note
+`batch_size` configuration will be overriden by `chunk_size` kwargs if provided
+````
+
+```{tip}
+You can read more about parallel bulk config and their default values [here](https://github.com/opensearch-project/opensearch-py/blob/main/opensearchpy/helpers/actions.py)
+```
+
+### Vector search with filter query
+
+One can perform Approximate Nearest Neighbor Search and pre-filter results using a filter query .
+
+Consider Documents with embeddings `[0,0,0]` up to `[9,9,9]` where the document with embedding `[i,i,i]`
+has as tag `price` with value `i`. We can create such example with the following code:
+
+```python
+from docarray import Document, DocumentArray
+import numpy as np
+
+n_dim = 3
+
+da = DocumentArray(
+    storage='opensearch',
+    config={'n_dim': n_dim, 'columns': {'price': 'int'}, 'distance': 'l2'},
+)
+
+with da:
+    da.extend(
+        [
+            Document(id=f'r{i}', embedding=i * np.ones(n_dim), tags={'price': i})
+            for i in range(10)
+        ]
+    )
+
+print('\nIndexed Prices:\n')
+for embedding, price in zip(da.embeddings, da[:, 'tags__price']):
+    print(f'\tembedding={embedding},\t price={price}')
+```
+
+Consider we want the nearest vectors to the embedding `[8. 8. 8.]`, with the restriction that
+prices must follow a filter. As an example, let's consider that retrieved documents must have `price` value lower
```

Comment:
Please ensure all occurrences of `Document` or `DocumentArray` are capitalized (since they are a class in DocArray)